### PR TITLE
Dami/hex no std

### PR DESCRIPTION
**Pull request by dharjeezy**
The protocol-ink poseidon contract could not compile successfully due to std feature issues. This is a slight change to enable the poseidon contract compile successfully.

### CODE REVIEW

File: arkworks-gadgets.iml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shekohex) [inline arkworks-gadgets.iml:1]: Please add this file to .gitignore.

File: arkworks-utils/Cargo.toml [hunk 2]
```diff
@@ -11,12 +11,12 @@ homepage = "https://webb.tools"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-hex = "0.4"
+hex = {version = "0.4", default-features = false}
```
Comment (shekohex) [inline arkworks-utils/Cargo.toml:14]:
```suggestion
hex = { version = "0.4", default-features = false }
```

File: arkworks-utils/src/lib.rs [hunk 3]
```diff
@@ -19,8 +19,10 @@ pub enum Curve {
 }
 
 pub fn decode_hex(s: &str) -> Result<Bytes, FromHexError> {
+	let mut bytes = ark_std::vec![0u8; s.len() / 2];
 	let s = &s[2..];
-	hex::decode(s)
+	hex::decode_to_slice(s, &mut bytes as &mut [u8]);
+	Ok(bytes.to_vec())
```
Comment (shekohex) [inline arkworks-utils/src/lib.rs:25]:
```suggestion
	Ok(bytes.into())
```

**Review by shekohex — CHANGES_REQUESTED** [review]
Looks good.

A few more small changes, before we can merge this.

File: arkworks-gadgets.iml (hunk 1 above)
Comment (dharjeezy) [reply arkworks-gadgets.iml:1]: I have added it

Comment (shekohex) [reply arkworks-gadgets.iml:1]: But it is still there, try using `git rm --cache arkworks-gadgets.iml` for example.

Comment (dharjeezy) [reply arkworks-gadgets.iml]: Done.

**Comment by drewstone** [comment]
Tests and CI need to be updated/fixed @dharjeezy

**Comment by dharjeezy** [comment]
Closing this because it has old commits. New PR here https://github.com/webb-tools/arkworks-gadgets/pull/216

**Pull request closed without merging**